libaleth-interpreter/VM.cpp

```diff
@@ -657,7 +655,9 @@ void VM::interpretCases()
             ON_OP();
             updateIOGas();
 
-            m_SPP[0] = u2s(m_SP[0]) < u2s(m_SP[1]) ? 1 : 0;
+            bool lhs_neg = static_cast<bool>(m_SP[0] >> 255);
+            bool rhs_neg = static_cast<bool>(m_SP[1] >> 255);
+            m_SPP[0] = (lhs_neg ^ rhs_neg) ? lhs_neg : m_SP[0] < m_SP[1];
```

Comment:
I see what you did here, but bitwise xor operator on `bool`s looks weird (I think it's implicitly converted back to integer), so I think `lhs_neg` and `rhs_neg` should be integers (maybe `uint8_t`?)

also names should be camelCase and should be `const`

Reply:
Fully agree that it looks weird. Its also not necessary -- I think the best way to avoid the implicit conversions bool -> int -> bool would be to replace the bitwise XOR operator '`^`' by the not-equal-to operator '`!=`'(since we already have boolean operands, this represents the desired "logical XOR" operation)? This should also express more clearly the intended condition "if the two operands have different signs", and we got rid of any implicit bool <-> int conversions.
Will change that and the naming (camelCase) plus `const` later if you don't have any concerns.